### PR TITLE
DEPENDENCY: upgrade to ember-cli & ember v3.0

### PR DESCRIPTION
**Pull request by toranb**
This includes everything except the qunit acceptance/unit test migration (doing that in the next pull request to keep the delta tiny)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,22 +27,27 @@
...
+    "ember-cli-uglify": "^2.0.0",
     "ember-disable-prototype-extensions": "^1.1.2",
     "ember-export-application-global": "^2.0.0",
     "ember-load-initializers": "^1.0.0",
+    "ember-maybe-import-regenerator": "^0.1.6",
     "ember-resolver": "^4.0.0",
-    "ember-source": "~2.15.0",
+    "ember-source": "~3.0.0",
+    "ember-try": "^0.2.23",
+    "ember-uuid": "^1.0.0",
```
Comment (snewcomer) [inline package.json]: ah this moved!  👍

**Review by snewcomer — COMMENTED** [review]
Looks good from the outside 🥇

**Comment by toranb** [comment]
@snewcomer after some playing around I decided to skip pulling in `ember-cli-uuid` /or `ember-uuid` to avoid relying on another 3rd party dependency -thoughts?

**Comment by snewcomer** [comment]
Seems like it is ok!  Never had a problem with that uuid module before 👍

**Pull request merged**